tests/test_tfe.py

```diff
@@ -167,26 +176,68 @@ def circuit(reused_param, other_param):
                 zero_state = np.array([1.,0.])
                 final_state = (Rx @ Rz @ Rx @ zero_state)
                 y_true = expZ(final_state)
-                self.assertAlmostEqual(y_eval, y_true, delta=self.tol)
 
-    def test_qnode_array_parameters(self):
-        "Test that QNode can take arrays as input arguments, and that they interact properly with TensorFlow."
-        self.logTestName()
+                assert np.allclose(y_eval, y_true, atol=tol, rtol=0)
+
+    def test_qnode_array_parameters_scalar_return(self, qubit_device_1_wire, tol):
+        """Test that QNode can take arrays as input arguments, and that they interact properly with TensorFlow."""
 
-        @qml.qnode(self.dev1, interface='tfe')
-        def circuit_n1s(dummy1, array, dummy2):
+        @qml.qnode(qubit_device_1_wire, interface='tfe')
+        def circuit(dummy1, array, dummy2):
             qml.RY(0.5 * array[0,1], wires=0)
             qml.RY(-0.5 * array[1,1], wires=0)
             return qml.expval(qml.PauliX(0))  # returns a scalar
 
-        @qml.qnode(self.dev1, interface='tfe')
-        def circuit_n1v(dummy1, array, dummy2):
+        grad_target = (np.array(1.), np.array([[0.5,  0.43879, 0], [0, -0.43879, 0]]), np.array(-0.4))
+        cost_target = 1.03257
+
+        args = (tfe.Variable(0.46), tfe.Variable([[2., 3., 0.3], [7., 4., 2.1]]), tfe.Variable(-0.13))
+
+        def cost(x, array, y):
+            c = tf.cast(circuit(tf.constant(0.111), array, tf.constant(4.5)), tf.float32)
+            if c.shape != tf.TensorShape([]):
```

Comment:
Since these three circuits are now split up into separate tests, there's no need for this conditional here, no?

Reply:
You're right! I'll change that.